omnibus/omnibus-test.sh

```diff
@@ -152,6 +152,18 @@ elif [[ -d /usr/local/etc/sudoers.d ]]; then
   sudo chmod 440 "/usr/local/etc/sudoers.d/$(id -un)-preserve_path"
 fi
 
+# Remove the cloud zypper repos before running functional tests
+if [[ -f /etc/SuSE-release ]] || [[ -f /etc/os-release ]] && [[ ! $(grep s390x /etc/SuSE-release /etc/os-release) ]]; then
+  if [[ -f /etc/os-release ]] && [[ ! $(grep -i suse  /etc/os-release) ]]; then
```

Comment:
this conditional seems a bit awkward and potentially brittle and could definitely use a comment or two at least.

- the s390x side of the conditional seems to be testing for our testing infrastructure rather than testing the zypper config that makes those nodes different.  can we get so that we're not treating s390x as a special case?

- since you've got mixed || and &&'s in the first conditional you should really use parens there for clarity

- the sense of the comparisons should be flipped around so that the `else` block becomes the body of the `if` block, the implicit negation is making it harder to read.

it looks like what this says is that if you have /etc/SuSE-release but not /etc/os-release" that you always prune the zypper repos, but that if you have /etc/os-release and it is suse (or really not-not-suse) and is not 390x then you prune it as well?

```
if [[ -f /etc/SuSE-release ]] || ( [[ -f /etc/os-release ]] && [[ ! $(grep s390x /etc/SuSE-release /etc/os-release) ]] && [[ $(grep -i suse  /etc/os-release) ]]; then
    echo "Removing zypper repos for Amazon SLES nodes"
    for i in $(zypper lr |grep -v "Alias" |awk -F "|" '{print $2}'); do
      sudo zypper rr $i
    done
fi
```

maybe that, unless i'm reading it wrong...

Reply:
@lamont-granquist I simplified the conditions to filter out s390x nodes. Also added code comments.

Reply:
https://buildkite.com/chef/chef-chef-master-omnibus-adhoc/builds/87 in progress to test these changes